src/main/java/heronarts/lx/effect/SparkleEffect.java

```diff
@@ -31,13 +31,11 @@
 
   public final SparklePattern.Engine engine = new SparklePattern.Engine(model);
 
-  public final CompoundParameter amount =
-    new CompoundParameter("Amount", 1)
-    .setDescription("Amount of sparkle to apply");
+
 
   public SparkleEffect(LX lx) {
     super(lx);
-    addParameter("amount", this.amount);
+    addParameter("amount", engine.amount1);
```

Comment:
Renaming this to `amount1` feels pretty clunky, and the parameter was intentionally on `SparkleEffect` because it is only registered as a parameter to `SparkleEffect` using the `addParameter` method, which makes it show up as a UI control for the effect, not the pattern. Hence, the parameter is declared and lives in the effect class.

I can see how the amount-passing is awkward here though, so have changed that to act as a passed parameter: https://github.com/heronarts/LX/commit/a05cfd14a14cf52da254c417fd33591d163dd873